gdk-pixbuf/all: Add 'with_introspection' option, fix libpng flag, glib bump and fix inconsistent quotes.

sh0 opened the pull request:
Specify library name and version:  **gdk-pixbuf/2.42.4**

~~Disable introspection deterministically as it does not seem to work due to `g-ir-scanner` bugs. Similar to PR #6709.~~
Edit: I have added `with_introspection` option instead of statically disabling introspection.

Avoid renaming `libpng.pc` when `with_libpng` flag is not enabled since the pkg-config file would not be there in that case.

Also bump glib version to latest.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

ericLemanissier commented:
As @Croydon wrote on another PR, please report the issue this intends to solve, and mark this PR as draft, so that we have a chance to fix the leg before removing it.

sh0 commented:
As with #6709 I have now added `with_introspection` option instead of statically disabling introspection. By default the flag is disabled.

I have created issue #6727 to track progress of compiling with introspection enabled.

Due to growing size of the PR I split other changes to separate commits. Also added a commit to make quotes consistent.

ericLemanissier reviewed (approved):
Thanks!
